huaweicloud/services/waf/data_source_huaweicloud_waf_dedicated_domains.go

```diff
@@ -0,0 +1,211 @@
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"website_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+	return &sc
+}
+
+func datasourceDedicatedDomainsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getWAFDedicatedDomains: Query WAF dedicated domains
+	var (
+		getWAFDedicatedDomainsHttpUrl = "v1/{project_id}/premium-waf/host"
+		getWAFDedicatedDomainsProduct = "waf"
+	)
+	getWAFDedicatedDomainsClient, err := cfg.NewServiceClient(getWAFDedicatedDomainsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating WAF client: %s", err)
+	}
+
+	getWAFDedicatedDomainsPath := getWAFDedicatedDomainsClient.Endpoint + getWAFDedicatedDomainsHttpUrl
+	getWAFDedicatedDomainsPath = strings.ReplaceAll(getWAFDedicatedDomainsPath, "{project_id}",
+		getWAFDedicatedDomainsClient.ProjectID)
+	getWAFDedicatedDomainsPath += buildWAFDedicatedDomainsQueryParams(d)
+
+	getWAFDedicatedDomainsResp, err := pagination.ListAllItems(
+		getWAFDedicatedDomainsClient,
+		"page",
+		getWAFDedicatedDomainsPath,
+		&pagination.QueryOpts{MarkerField: ""})
+
+	if err != nil {
+		return diag.Errorf("error retrieving dedicated domains, %s", err)
+	}
+
+	listWAFDedicatedDomainsRespJson, err := json.Marshal(getWAFDedicatedDomainsResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var listWAFDedicatedDomainsRespBody interface{}
+	err = json.Unmarshal(listWAFDedicatedDomainsRespJson, &listWAFDedicatedDomainsRespBody)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	dataSourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(dataSourceId)
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("domains", flattenListDedicatedDomainsBody(listWAFDedicatedDomainsRespBody)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenListDedicatedDomainsBody(resp interface{}) []interface{} {
+	if resp == nil {
+		return nil
+	}
+	curJson := utils.PathSearch("items", resp, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+	rst := make([]interface{}, 0, len(curArray))
+	for _, v := range curArray {
+		rst = append(rst, map[string]interface{}{
+			"id":                    utils.PathSearch("id", v, nil),
+			"domain":                utils.PathSearch("hostname", v, nil),
+			"policy_id":             utils.PathSearch("policyid", v, nil),
+			"enterprise_project_id": utils.PathSearch("enterprise_project_id", v, nil),
+			"description":           utils.PathSearch("description", v, nil),
+			"protect_status":        utils.PathSearch("protect_status", v, nil),
+			"website_name":          utils.PathSearch("web_tag", v, nil),
+			"access_status":         utils.PathSearch("access_status", v, nil),
+			"pci_3ds":               utils.PathSearch("pci_3ds", v, nil),
```

Comment:
```suggestion
			"pci_3ds":               utils.PathSearch("flag.pci_3ds", v, nil),
```
same question to field `pci_dss`.